#!/usr/bin/env python3
PACKAGE = "val_soft_estop_monitor"

from dynamic_reconfigure.parameter_generator_catkin import *

# initialize parameter generator to define parameters
gen = ParameterGenerator()

# add parameters to generator
gen.add("debug",                   bool_t,   0, "Debug for printing",                                                       False)
gen.add("monitor_hands",           bool_t,   0, "Flag to monitor hands",                                                    True)
gen.add("ee_hand_pos_delta_limit", double_t, 0, "Limit for distance between desired and actual hand position (meters)",     0.5,  0.0, 0.5)  # TODO default, min, max values?
gen.add("ee_hand_ori_delta_limit", double_t, 0, "Limit for distance between desired and actual hand orientation (radians)", 3.15, 0.0, 3.15) # TODO default, min, max values?
gen.add("monitor_head",            bool_t,   0, "Flag to monitor head",                                                     True)
gen.add("ee_head_pos_delta_limit", double_t, 0, "Limit for distance between desired and actual head position (meters)", 	0.5,  0.0, 0.5)  # TODO default, min, max values?
gen.add("ee_head_ori_delta_limit", double_t, 0, "Limit for distance between desired and actual head orientation (radians)", 2.0,  0.0, 3.15) # TODO default, min, max values?

# generate necessary files
exit(gen.generate(PACKAGE, "val_soft_estop_monitor", "EndEffectorMonitorParams"))
